modules/openapi-generator/src/main/resources/typescript-nestjs/api.service.mustache

```diff
@@ -0,0 +1,237 @@
+{{>licenseInfo}}
+/* tslint:disable:no-unused-variable member-ordering */
+
+import { HttpService, Inject, Injectable }                      from '@nestjs/common';
+import { AxiosResponse } from 'axios';
+import { Observable }                                        from 'rxjs';
+{{#imports}}
+import { {{classname}} } from '../{{filename}}';
+{{/imports}}
+import { Configuration }                                     from '../configuration';
+import { COLLECTION_FORMATS }                     from '../variables';
+{{#withInterfaces}}
+import { {{classname}}Interface }                            from './{{classFilename}}Interface';
+{{/withInterfaces}}
+
+{{#operations}}
+
+{{#description}}
+/**
+ * {{&description}}
+ */
+{{/description}}
+@Injectable()
+{{#withInterfaces}}
+export class {{classname}} implements {{classname}}Interface {
+{{/withInterfaces}}
+{{^withInterfaces}}
+export class {{classname}} {
+{{/withInterfaces}}
+
+    protected basePath = '{{{basePath}}}';
+    public defaultHeaders = new Map()
+    public configuration = new Configuration();
+
+    constructor(protected httpClient: HttpService, @Inject('BASE_PATH') basePath: string, configuration: Configuration) {
```

Comment:
`BASE_PATH` DI token will also be ambiguous if you have multiple generated clients in the project. We probably want to give it a more specific and unique name, like `_OPENAPI_GENERATED_{{apiName}}_BASE_PATH`, so it doesn't collide with any DI tokens that the users might already have in their Nest apps.

Reply:
Since it's a brand new generator with no need for backward compatibility, I would suggest getting rid of a dedicated "BASE_PATH" injection token altogether - in favor of Configuration (which also includes basePath)

Reply:
@amakhrov I agree with that makes sense